source/dofs/dof_handler_policy.cc

```diff
@@ -2056,10 +2083,15 @@ namespace internal
             if (!cell->is_artificial())
               if ((subdomain_id == numbers::invalid_subdomain_id) ||
                   (cell->subdomain_id() == subdomain_id))
-                next_free_dof =
-                  Implementation::distribute_dofs_on_cell(dof_handler,
-                                                          cell,
-                                                          next_free_dof);
+                {
+                  next_free_dof =
+                    Implementation::distribute_dofs_on_cell(dof_handler,
+                                                            cell,
+                                                            next_free_dof);
+                  next_free_dof =
+                    Implementation::distribute_non_local_dofs_on_cell(
+                      dof_handler, cell, next_free_dof);
```

Comment:
I think in the implementation above, you stuck the non-local DoF numbering into `distribute_dofs_on_cell`. But then you don't need to call this function again here.